### ISSUE TITLE
Multiple RGW Realms

### ISSUE THREAD

**Issue by jon-rei** [galexrt/extended-ceph-exporter#1]
Hi there, 
first of all, thanks a lot for continuing to provide support for this exporter.
Is it possible to use multiple rgw realms with one exporter deployment? In my setup I'm using multiple rgw realms and to monitor all of them I would now need to deploy the helm chart multiple times. Is it possible to use only one exporter with multiple rgw realm configurations? As far as I can see from the config, you can only configure one rgw host.

**Comment by galexrt** [comment]
Mhh, good question, I would need to check how easy it would be to implement that in the exporter.

**Comment by galexrt** [comment]
@jon-rei From the current "architecture" the exporter simply creates one RGW admin client, so it might be enough to take in a list of RGW hosts and credentials, and have each "metrics collector" run per RGW admin client connection so the metrics would be per "realm."

Does that sound "sane" from your perspective?

**Comment by jon-rei** [comment]
Yes, that actually sounds like a good solution and would be better than having to run the helm chart multiple times.

**Comment by galexrt** [comment]
@jon-rei Could you please take a look at the `realms.example.yaml` and `values.yaml` file changes in the #2 PR

I still need to test the changes, but I would currently want to settle on a config "format" for the so called "multi realm mode" for the exporter.

**Comment by jon-rei** [comment]
Thanks for working on this. I took a look at the PR and as far as I understood it to enable multi realm mode I would then need to pass the access/secret key either through helm values or create an external secret and pass it to the deployment through a volume mount right?
I guess this would be suitable for how I currently deploy the exporter.

**Comment by galexrt** [comment]
@jon-rei You must provide an external secret and pass that to the deployment. I have added a `extraObjects` to the helm chart so you could also have the helm chart create the secret or "manually."

I have updated the README with a passage about what if the multi realm mode is enabled.

**Comment by galexrt** [comment]
I have tested the RGW multi mode changes with my RGW cluster and it seems to work fine.

I'm going to create a new release `1.3.0` of extended-ceph-exporter soon with his feature added, and the Helm chart version of `1.4.1` will be available soon.

**Comment by bumarcell** [comment]
Multi mode doesn't seem to work for me;
I added realms.yaml with two realms (neither is called default), but the metrics are still only reporting for realm=default

I also don't see the the multimode values neither in the args nor in the config secret

**Comment by bumarcell** [comment]
@galexrt I believe the error is [here](https://github.com/galexrt/extended-ceph-exporter/blob/main/charts/extended-ceph-exporter/templates/secrets.yaml#L8) where `multiModeEnabled` should be replaces with `multiMode.enabled`

**Comment by galexrt** [comment]
> @galexrt I believe the error is [here](https://github.com/galexrt/extended-ceph-exporter/blob/main/charts/extended-ceph-exporter/templates/secrets.yaml?rgh-link-date=2024-08-12T15%3A11%3A28Z#L8) where `multiModeEnabled` should be replaced with `multiMode.enabled`

Yes, that seems like the culprit. Would be up to submit a PR to fix this?
